saleor/payment/gateways/stripe/forms.py

```diff
@@ -0,0 +1,75 @@
+from django import forms
+from django.forms.utils import flatatt
+from django.forms.widgets import HiddenInput
+from django.utils.html import format_html
+from django.utils.translation import pgettext_lazy
+
+from ...forms import PaymentForm
+from .utils import get_amount_for_stripe
+
+CHECKOUT_SCRIPT_URL = 'https://checkout.stripe.com/checkout.js'
+CHECKOUT_DESCRIPTION = pgettext_lazy(
+    'Stripe payment gateway description', 'Total payment')
+
+
+class StripeCheckoutWidget(HiddenInput):
+
+    def __init__(self, payment, gateway_params, *args, **kwargs):
+        attrs = kwargs.get('attrs', {})
+        kwargs['attrs'] = {
+            'class': 'stripe-button',
+            'src': CHECKOUT_SCRIPT_URL,
+            'data-key': gateway_params.get('public_key'),
+            'data-amount': get_amount_for_stripe(
+                payment.total, payment.currency),
+            'data-name': gateway_params.get('store_name'),
+            'data-image': gateway_params.get('store_image'),
+            'data-description': CHECKOUT_DESCRIPTION,
+            'data-currency': payment.currency,
+            'data-locale': gateway_params.get('locale'),
+            'data-allow-remember-me': 'true' if gateway_params.get(
+                'remember_me') else 'false',
+            'data-billing-address': 'true' if gateway_params.get(
```

Comment:
Shouldn't we provide the billing address by default to prevent fraud and facilitate the dispute process? (even if the client can alter this data since it is client side only).

Reply:
Sadly, stripe.js or checkout.js of stripe doesn't support pre-fill the billing or shipping address. Enable this will add an extra step for the user to input billing/shipping address during checkout. The user actually has already provide the address in saleor's checkout process, and re-input this feels very annoying. That's why these options are falsy by default.